modules/tinymce/src/plugins/table/main/ts/api/Options.ts

```diff
@@ -0,0 +1,277 @@
+  });
+
+  registerOption('table_resize_bars', {
+    processor: 'boolean',
+    default: true
+  });
+
+  registerOption('table_tab_navigation', {
+    processor: 'boolean',
+    default: true
+  });
+
+  registerOption('table_cell_advtab', {
+    processor: 'boolean',
+    default: true
+  });
+
+  registerOption('table_row_advtab', {
+    processor: 'boolean',
+    default: true
+  });
+
+  registerOption('table_advtab', {
+    processor: 'boolean',
+    default: true
+  });
+
+  registerOption('table_appearance_options', {
+    processor: 'boolean',
+    default: true
+  });
+
+  registerOption('table_grid', {
+    processor: 'boolean',
+    // Table grid relies on hover, which isn't available on touch devices so use the dialog instead
+    default: !Env.deviceType.isTouch()
+  });
+
+  registerOption('table_style_by_css', {
+    processor: 'boolean',
+    default: false
+  });
+
+  registerOption('table_cell_class_list', {
+    processor: 'array',
+    default: []
+  });
+
+  registerOption('table_row_class_list', {
+    processor: 'array',
+    default: []
+  });
+
+  registerOption('table_class_list', {
+    processor: 'array',
+    default: []
+  });
+
+  registerOption('table_toolbar', {
+    processor: 'string',
+    default: defaultTableToolbar
+  });
+
+  registerOption('table_use_colgroups', {
+    processor: 'boolean',
+    default: false
+  });
+
+  registerOption('table_header_type', {
+    processor: (value) => {
+      const valid = Arr.contains([ 'section', 'cells', 'sectionCells', 'auto' ], value);
+      return valid ? { value, valid } : { valid: false, message: 'Must be one of: section, cells, sectionCells or auto.' };
+    },
+    default: 'section'
+  });
+
+  registerOption('table_column_resizing', {
+    processor: (value) => {
+      const valid = Arr.contains([ 'preservetable', 'resizetable' ], value);
+      return valid ? { value, valid } : { valid: false, message: 'Must be preservetable, or resizetable.' };
+    },
+    default: 'preservetable'
+  });
+
+  registerOption('table_clone_elements', {
+    processor: (value) => {
+      if (Type.isString(value)) {
+        return { value: value.split(/[ ,]/), valid: true };
+      } else if (Type.isArrayOf(value, Type.isString)) {
+        return { value, valid: true };
+      } else {
+        return { valid: false, message: 'Must be a string or an array of strings.' };
+      }
+    }
+  });
+
+  registerOption('table_background_color_map', {
+    processor: 'array',
+    default: []
+  });
+
+  registerOption('table_border_color_map', {
+    processor: 'array',
+    default: []
+  });
+};
+
+const getTableSizingMode = option<TableSizingMode>('table_sizing_mode');
+const getTableBorderWidths = option<UserListItem[]>('table_border_widths');
+const getTableBorderStyles = option<UserListValue[]>('table_border_styles');
+const getDefaultAttributes = option<StringMap>('table_default_attributes');
+const hasTableResizeBars = option<boolean>('table_resize_bars');
```

Comment:
I keep seeing this and thinking it's an optional boolean. I don't think we need to change it, I just wanted to make that clear 😂